Neos.Flow/Classes/ObjectManagement/Configuration/ConfigurationBuilder.php

```diff
@@ -313,30 +313,25 @@ protected function parseScope($value)
      * @return integer The autowiring option translated into one of Configuration::AUTOWIRING_MODE_*
      * @throws InvalidObjectConfigurationException if an invalid option has been specified
      */
-    protected static function parseAutowiring($value)
+    protected static function parseAutowiring(mixed $value): int
     {
-        switch ($value) {
-            case true:
-            case Configuration::AUTOWIRING_MODE_ON:
-                return Configuration::AUTOWIRING_MODE_ON;
-            case false:
-            case Configuration::AUTOWIRING_MODE_OFF:
-                return Configuration::AUTOWIRING_MODE_OFF;
-            default:
-                throw new InvalidObjectConfigurationException('Invalid autowiring declaration', 1283866757);
-        }
+        return match ($value) {
```

Comment:
you might have uncovered something that is better hidden: https://github.com/neos/flow-development-collection/issues/3307

will revert this method so its a cleanup and not a bugfix :O